src/js/actions/LibraryActions.js

```diff
@@ -141,21 +120,87 @@ const refresh = () => {
                 if (docs.length === 0) {
                     return utils.getMetadata(filePath);
                 }
-                return docs[0];
+                return null;
             }).then((track) => {
+                if(track === null) return;
                 return app.models.Track.insertAsync(track);
             }).then(() => {
-                const percent = parseInt(addedFiles * 100 / totalFiles);
-                AppActions.settings.refreshProgress(percent);
+                refreshProgress(addedFiles, totalFiles);
                 addedFiles++;
             });
         }, { concurrency: fsConcurrency });
     }).then(() => {
-        AppActions.library.load();
+        AppActions.library.load(); // TODO Reload the library at the end of this
         dispatchEnd();
     }).catch((err) => {
         console.warn(err);
     });
+
+    // TODO progressive loading in the store, don't freeze the app, able to add files/folders when scanning
+};
+
+const refreshProgress = (processed, total) => {
+    store.dispatch({
+        type : AppConstants.APP_LIBRARY_REFRESH_PROGRESS,
+        processed,
+        total,
+    });
+};
+
+const removeFromLibrary = async (tracksIds) => {
+    // not calling await on it as it calls the synchonous message box
+    dialog.showMessageBox(app.browserWindows.main, {
+        buttons: [
+            'Cancel',
+            'Remove'
+        ],
+        title: 'Remove tracks from library?',
+        message: `Are you sure you want to remove ${tracksIds.length} element(s) from your library?`,
+        type: 'warning'
+    }, (result) => {
+        if(result === 1) {
```

Comment:
I'd declare this constant somewhere in top scope so we can compare to a named constant instead of magic numbers.